tests/test_save_read_dump.rs

```diff
@@ -0,0 +1,67 @@
+/// Cache config should be injected to the handler!
+/// This is to enable client to configure things dynamically.
+
+/// if the value of dir is /tmp, then the expected response to CONFIG GET dir is:
+/// *2\r\n$3\r\ndir\r\n$4\r\n/tmp\r\n
+mod common;
+use crate::common::{array, keys_command, null_response, ok_response, save_command, set_command, set_command_with_expiry, start_test_server};
+use common::{integration_test_config, TestStreamHandler};
+use std::env;
+use std::time::{SystemTime, UNIX_EPOCH};
+use tokio::net::TcpStream;
+
+#[tokio::test]
+async fn test() {
+    // GIVEN
+    let test_file_name = create_unique_file_name("test_save_dump");
+    env::set_var("dbfilename", &test_file_name);
+
+    let config = integration_test_config().await;
+    let _ = start_test_server(config).await;
+
+    let mut client_stream = TcpStream::connect(config.bind_addr()).await.unwrap();
+    let mut h: TestStreamHandler = client_stream.split().into();
+
+    // WHEN
+    // set without expiry time
+    h.send(set_command("foo", "bar").as_slice()).await;
+    assert_eq!(h.get_response().await, ok_response());
+    // set with expiry time
+    // h.send(b"*5\r\n$3\r\nSET\r\n$4\r\nfoo2\r\n$4\r\nbar2\r\n$2\r\npx\r\n$10\r\n9999999999\r\n'").await;
+    h.send(set_command_with_expiry("foo2", "bar2", 9999999999).as_slice()).await;
+    assert_eq!(h.get_response().await, ok_response());
+    // check keys
+    h.send(keys_command().as_slice()).await;
+    assert_eq!(h.get_response().await, array(vec!["foo", "foo2"]));
+    // save
+    h.send(save_command().as_slice()).await;
+
+    // THEN
+    assert_eq!(h.get_response().await, null_response());
+
+    // wait for the file to be created
+    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
+
+    // start another instance
+    let config = integration_test_config().await;
```

Comment:
Guess at this point we should consider taking control over where to find the file from. 

Plus, if there is other processes, the test of which pertains to dump file, it may collide?

Reply:
```rust
let test_file_name = create_unique_file_name("test_save_dump");
env::set_var("dbfilename", &test_file_name);
``` 
I'm not sure if I fully understand, but since we're using file names generated from nanosecond-level timestamps, we are controlling the files used in this test. However it is true that if another process tries to create and read the same file at the exact same time, a collision might occur. But I think it's unlikely to happen, so I wrote it like this. Do you think this is okay?

Reply:
@echo724 Not really. it would set global environment variable which will affect other test suites. 

Perhaps, you can simply change input parameter of `integration_test_config` and add `set_xxx` method on `Config` so you can set config values dynamically?